### PR TITLE
[CPDLP-2940] Add courses migrator

### PR DESCRIPTION
**Pull request by leandroalemao**
### Context

Ticket: [CPDLP-2962](https://dfedigital.atlassian.net/browse/CPDLP-2962)

We would like to migrate existing courses data into NPQ from ECF.

### Changes proposed in this pull request

 Add logic to migrate ECF npq courses data into NPQ reg app.

image

[CPDLP-2962]: https://dfedigital.atlassian.net/browse/CPDLP-2962?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: config/environments/migration.rb [hunk 1]
```diff
@@ -4,6 +4,13 @@
   config.log_level = :debug
   config.ssl_options = { redirect: { exclude: ->(request) { request.path.include?("/healthcheck") } } }
 
+  config.action_controller.perform_caching = true
```
Comment (ethax-ross) [inline config/environments/migration.rb:7]: Does the migration environment not have Redis already setup for caching? I think we should keep it as close to prod as possible

Comment (cwrw) [reply config/environments/migration.rb:7]: we don't have Redis in NPQ reg, adding a ticket to add it

File: app/models/course.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class Course < ApplicationRecord
+  validates :name, presence: true
```
Comment (cwrw) [inline app/models/course.rb:2]: ✂️ as this has been added in another pr :)

File: spec/models/course_spec.rb [hunk 3]
```diff
@@ -1,6 +1,10 @@
 require "rails_helper"
 
 RSpec.describe Course do
+  describe "validations" do
```
Comment (cwrw) [inline spec/models/course_spec.rb:4]: ✂️

**Review by cwrw — COMMENTED** [review]
looks good, however keeps failing when I try to test on migration, will try running tomorrow

**Review by cwrw — APPROVED** [review]
tested on migration, thanks @leandroalemao

**Pull request merged**